src/blr/cli/tagfastq.py

```diff
@@ -89,20 +90,21 @@ def main(args):
     logger.info("Finished")
 
 
-def search_bc(iterator: iter, header: str, cache: dict, maxiter: int = 10):
-    # Check it header is stored in cache. If not move forward on step in iterator at look again.
-    iteration = 0
-    while header not in cache and iteration < maxiter:
-        iteration += 1
-        # Progress iterator
-        try:
-            cache.update(next(iterator))
-        except StopIteration:
-            break
+def search_barcode(uncorrected_barcodes, header: str, cache, maxiter: int = 10):
+    if header in cache:
+        return cache.pop(header)
+
+    for iteration, header_barcode_pair in enumerate(uncorrected_barcodes):
```

Comment:
You could avoid the extra `iteration` variable by using `islice` here (`from itertools import islice`):
```
for header_barcode_pair in islice(uncorrected_barcodes, maxiter):
```
Then also the `if iteration >= maxiter:` below isn’t needed.

Reply:
Sweet! I will add that.